waspc/data/Generator/templates/sdk/wasp/server/operations/wrappers.ts

```diff
@@ -168,6 +168,13 @@ type AuthenticatedOperationArgsFor<Op extends GenericAuthenticatedOperationDefin
  * @template Output The type of the operation's return value.
  */
 type AuthenticatedOperation<Input, Output> =
+  IfAny<
+    Input,
+    (args: any, context: { user: AuthUser }) => Promise<Output>,
```

Comment:
I've noticed some repetition here:
```diff
+(args: any, context: { user: AuthUser }) => Promise<Output>
-(args: unknown, context: { user: AuthUser }) => Promise<Output>
```


And below:
```diff
+(args: any) => Promise<Output>
-(args: unknown) => Promise<Output>
```

Would it make sense to create some generics like: 
```ts
type UnauthenticatedOperationFnWithArgs<Args, Output> = (args: Args) => Promise<Output>
type AuthenticatedOperationFnWithArgs<Args, Output> = (args: Args, context: { user: AuthUser }) => Promise<Output>
```

Maybe that's too much?

Reply:
I thought about it, changed it back and forth and ended up with what I have now on the account of `any` resolving in the same generic where we check for it.

But now, when you suggested it, I've changed my mind again. So I think I'll do it. Thanks.

Reply:
I couldn't do it.

The generic helpers you suggested messed up how the IDE displays the types:
![image](https://github.com/wasp-lang/wasp/assets/22752307/b6530fed-2f15-40b8-87b7-3ee9a594f57f)

Of course, I then tried fixing it with `Expand`.

On the server,  `Expand<IfAny<...>>` was working fine.
But on the client `Expand` started acting weird - adding it anywhere in the type hierarchy was causing weird type errors.

For example, doing this:
```typescript
type ClientOperation<Input, Output> =
  Expand<IfAny<
    Input,
    ClientOperationWithOptionalArgs<any, Output>,
    ClientOperationWithNonAnyInput<Input, Output>
  >
```
<details>
<summary>Caused a bunch of errors.</summary>

```
client/operations/queries/core.ts:41:45 - error TS2345: Argument of type 'QueryFor<BackendQuery>' is not assignable to parameter of type 'Query<Parameters<BackendQuery>[0], _Awaited<_ReturnType<BackendQuery>>>'.
  Type 'QueryFor<BackendQuery>' is not assignable to type 'Expand<IfAny<Parameters<BackendQuery>[0], ClientOperationWithOptionalArgs<any, _Awaited<_ReturnType<BackendQuery>>>, ClientOperationWithNonAnyInput<...>>>'.

41     const queryCacheKey = makeQueryCacheKey(queryFn as QueryFor<BackendQuery>, queryArgs)
                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

ext-src/Todo.tsx:44:61 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type '(GetResult<{ id: number; description: string; isDone: boolean; userId: number; }, unknown> & {})[]'.

44           <ToggleAllTasksButton disabled={!areThereAnyTasks(tasks)} />
                                                               ~~~~~

ext-src/Todo.tsx:50:27 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type '(GetResult<{ id: number; description: string; isDone: boolean; userId: number; }, unknown> & {})[]'.
  Type '{}' is missing the following properties from type '(GetResult<{ id: number; description: string; isDone: boolean; userId: number; }, unknown> & {})[]': length, pop, push, concat, and 35 more.

50         {areThereAnyTasks(tasks) && (
                             ~~~~~

ext-src/pages/Task.tsx:64:27 - error TS2339: Property 'id' does not exist on type 'unknown'.

64           <div> id: {task.id} </div>
                             ~~

ext-src/pages/Task.tsx:65:36 - error TS2339: Property 'description' does not exist on type 'unknown'.

65           <div> description: {task.description} </div>
                                      ~~~~~~~~~~~

ext-src/pages/Task.tsx:66:32 - error TS2339: Property 'isDone' does not exist on type 'unknown'.

66           <div> is done: {task.isDone ? 'Yes' : 'No'} </div>
                                  ~~~~~~

ext-src/pages/Task.tsx:67:47 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'GetResult<{ id: number; description: string; isDone: boolean; userId: number; }, unknown> & {}'.
  Type '{}' is missing the following properties from type 'GetResult<{ id: number; description: string; isDone: boolean; userId: number; }, unknown>': id, description, isDone, userId

67           <button onClick={() => toggleIsDone(task)}>
                                                 ~~~~

ext-src/pages/Task.tsx:68:27 - error TS2339: Property 'isDone' does not exist on type 'unknown'.

68             Mark as {task.isDone ? 'undone' : 'done'}
                             ~~~~~~

ext-src/pages/auth/Signup.tsx:46:43 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'number'.

46           <span>{getTotalTaskCountMessage(numTasks)}</span>
                                             ~~~~~~~~


Found 9 errors in 4 files.

Errors  Files
     1  client/operations/queries/core.ts:41
     2  ext-src/Todo.tsx:44
     5  ext-src/pages/Task.tsx:64
     1  ext-src/pages/auth/Signup.tsx:46
```

</details>

Which is pretty weird on its own, but especially weird since the tests aren't failing.

As soon as I removed `Expand`, everything was OK (aside from the display issues).

I tried using a simplified `ExpandFunction` that deals only with functions:
```typescript
type ExpandFunction<F> =
  F extends (...args: infer Args) => infer Ret
  ? (...args: Args) => Ret
  : never
```
That didn't work either.

Finally, after spending half a day of trying to figure this out, I gave up and decided to keep the duplication on both the server and the client (for consistency and because I don't know what's going on)

Reply:
We tried and it didn't work out, the current state is not too bad either 👍